docs/decisions/0048-agent-chat-serialization.md

```diff
@@ -0,0 +1,374 @@
+|---|---|
+|`ChatCompletionAgent`|`ChatCompletionAgent`|
+|`OpenAIAssistantAgent`|`OpenAIAssistantAgent`|
+|`ChatCompletionAgent` & `OpenAIAssistantAgent`|`ChatCompletionAgent` & `OpenAIAssistantAgent`|
+
+#### 2. **Enhanced:** Additional original agent types (channels) available in the restored chat.
+This shall also result in a full-fidelity restoration of of the original chat.
+Any new agent type (channel) will synchronize to the chat once restored (identical to adding a new agent type to a chat that is progress).
+
+|Source Chat|Target Chat|
+|---|---|
+|`ChatCompletionAgent`|`ChatCompletionAgent` & `OpenAIAssistantAgent`|
+|`OpenAIAssistantAgent`|`ChatCompletionAgent` & `OpenAIAssistantAgent`|
+
+#### 3. **Reduced:** A subset of original agent types (channels) available in the restored chat.
+This shall also result in a full-fidelity restoration of of the original chat to the available channels.  Introduction of a missing agent type (channel) post restoration will
+synchronize the channel to the current chat (identical to adding a new agent type to a chat that is progress).
+
+|Source Chat|Target Chat|
+|---|---|
+|`ChatCompletionAgent` & `OpenAIAssistantAgent`|`ChatCompletionAgent`|
+|`ChatCompletionAgent` & `OpenAIAssistantAgent`|`OpenAIAssistantAgent`|
+
+#### 4. **Empty:** No agents available in the restored chat.
+This shall result in an immediate exception (fail-fast) in order to strongly indicate that
+the chat has not been restored.  The chat may have agents added in order to attempt a successful restoration, or utilized on its own.  That is, the `AgentChat` instance isn't invalidated.
+
+#### 5. **Invalid:** Chat has already developed history or channels state.
+This shall result in an immediate exception (fail-fast) in order to strongly indicate that
+the chat has not been restored.  The chat may continue to be utilized as the `AgentChat` instance isn't invalidated.
+
+#### Notes:
+
+> Once restored, additional `Agent` instances may join the `AgentChat`, no different from any `AgentChat` instance.
+
+## Analysis
+
+#### Relationships:
+
+The relationships between any `AgentChat`, the `Agent` instances participating in the conversation, and the associated `AgentChannel` conduits are illustrated in the following diagram:
+
+![AgentChat Relationships](diagrams/agentchat-relationships.png)
+
+While an `AgentChat` manages a primary `ChatHistory`, each `AgentChannel` manages how that history is adapted to the specific `Agent` modality.  For instance, an `AgentChannel` for an `Agent` based on the Open AI Assistant API tracks the associated _thread-id_.  Whereas a `ChatCompletionAgent` manages an adapted `ChatHistory` instance of its own.
+
+This implies that logically the `AgentChat` state must retain the primary `ChatHistory` in addition to the appropriate state for each `AgentChannel`:
+
+#### Logical State:
+
+These relationships translate into the following logical state definition:
+
+![AgentChat State](diagrams/agentchat-state.png)
```

Comment:
Thoughs on additional serialization meta-data:

- **Language/platform**:  .Net, Python, Java.  Ideally the goal here is that the data is equivalent across platforms.
- **Save Date / When Created**: For audit
- **Build Version**: Not sure if this translates into anything useful
- **Serializer Version**: Could be useful for backwards compat (reading old saved chats, should the state evolve)
- **Other**: ???